bindings/python/scripts/clang_utils.py

```diff
@@ -0,0 +1,120 @@
+import inspect
+import clang.cindex as clang
+
+
+"""
+- `getmembers` from the inspect module triggers execution, which leads to errors when types are not satisfied.
+- So we define a custom function based on the one in the inspect module.
+- It replaces `getattr_static` with `getattr_static` to solve the issue.
+"""
+
+
+def getmembers_static(object, predicate=None):
+    """Return all members of an object as (name, value) pairs sorted by name.
+    Optionally, only return members that satisfy a given predicate."""
+    if inspect.isclass(object):
+        mro = (object,) + inspect.getmro(object)
+    else:
+        mro = ()
+    results = []
+    processed = set()
+    names = dir(object)
+    # :dd any DynamicClassAttributes to the list of names if object is a class;
+    # this may result in duplicate entries if, for example, a virtual
+    # attribute with the same name as a DynamicClassAttribute exists
+    try:
+        for base in object.__bases__:
+            for k, v in base.__dict__.items():
+                if isinstance(v, types.DynamicClassAttribute):
+                    names.append(k)
+    except AttributeError:
+        pass
+    for key in names:
+        # First try to get the value via getattr.  Some descriptors don't
+        # like calling their __get__ (see bug #1785), so fall back to
+        # looking in the __dict__.
+        try:
+            value = inspect.getattr_static(object, key)
+            # handle the duplicate key
+            if key in processed:
+                raise AttributeError
+        except AttributeError:
+            for base in mro:
+                if key in base.__dict__:
+                    value = base.__dict__[key]
+                    break
+            else:
+                # could be a (currently) missing slot member, or a buggy
+                # __dir__; discard and move on
+                continue
+        if not predicate or predicate(value):
+            results.append((key, value))
+        processed.add(key)
+    results.sort(key=lambda pair: pair[0])
+    return results
+
+
+# A list to ignore the functions/properties that causes segmentation errors.
+ignore_list = ["mangled_name", "get_address_space", "get_typedef_name", "tls_kind"]
```

Comment:
Any reason this is not inside the function on L61?

Reply:
Nope, shifted.